src/validator/rpc/abstract.ts

```diff
@@ -0,0 +1,86 @@
+import {NewEpochCallback, NewSlotCallback, RpcClient} from "./interface";
+import {Slot} from "../../types";
+import {IValidatorApi} from "../../rpc/api/validator";
+import {intDiv} from "../../util/math";
+import {SECONDS_PER_SLOT} from "../../constants";
+import {slotToEpoch} from "../../chain/stateTransition/util";
+import {IBeaconApi} from "../../rpc/api/beacon";
+
+
+export abstract class AbstractRpcClient implements RpcClient {
+
+  private currentSlot: Slot;
+
+  private currentEpoch: Slot;
+
+  private newSlotCallbacks: NewSlotCallback[] = [];
+  private newEpochCallbacks: NewEpochCallback[] = [];
+
+  private running = false;
+
+  public onNewEpoch(cb: NewEpochCallback): void {
```

Comment:
this is never called, so the loop on line 69 would never do anything

Reply:
What do you mean this is never called? Isn't validator suppose to on each new epoch look for duties in next epoch(lookadhead)?

Reply:
Sorry that was the most unclear message ever... lol

`this.onNewEpoch` is never called, thus on line 69:
```
      this.newEpochCallbacks.forEach((cb) => {
        cb(this.currentEpoch);
      });
```
will itterate over an empty array everytime, we might want to add another setTimeout to `startSlotCounting`

Reply:
Validator client will invoke rpcClient.onNewEpoch(cb); where it will provide his callback to listen on new epoch.

Reply:
oh understood, that makes sense !